src/main/java/org/broadinstitute/hellbender/tools/exome/hmm/CopyNumberTriStateHiddenMarkovModel.java

```diff
@@ -0,0 +1,354 @@
+
+    /**
+     * Valid range for the duplication event distribution center parameter.
+     */
+    public static final DoubleRange VALID_DUPLICATION_CENTER_RANGE = new DoubleRange(0, Double.MAX_VALUE);
+
+    /**
+     * Valid range for the intra-event target distance.
+     */
+    public static final DoubleRange VALID_MEAN_INTRA_EVENT_TARGET_DISTANCE_RANGE
+            = new DoubleRange(1, Double.MAX_VALUE);
+
+    /**
+     * Cached <code>log(p)</code> where <code>p</code> is the <i>event-rate</i>.
+     */
+    private final double logP;
+
+    /**
+     * Cached <code>log(1 - 2p)</code> where <code>p</code> is the <i>event-rate</i> .
+     */
+    private final double log1Minus2P;
+
+    /**
+     * Cached <code>log(q)</code> where <code>q</code> is the <i>inverse of the mean event size in targets</i>.
+     */
+    private final double logQ;
+
+    /**
+     * Cached <code>log(1 - q)</code> where <code>q</code> is the <i>inverse of the mean event size in targets</i>.
+     */
+    private final double log1MinusQ;
+
+    /**
+     * Cached <code>1/D</code> where <code>D</code> is the <i>average distance between targets in the same event</i>.
+     */
+    private final double invD;
+
+    /**
+     * Deletion emission distribution.
+     */
+    private final NormalDistribution deletionDistribution;
+
+    /**
+     * Duplication emission distribution.
+     */
+    private final NormalDistribution duplicationDistribution;
+
+    /**
+     * Holds the mean number of bps between targets in the same event.
+     */
+    private final double D;
+
+    /**
+     * Holds the event-rate per target.
+     */
+    private final double P;
+
+    /**
+     * Mean number of targets per event.
+     */
+    private final double T;
+
+    /**
+     * Creates a new model instance.
+     *
+     * @param eventRate the event-rate for this instance.
+     * @param meanNumberOfTargetsPerEvent the expectation for the number of targets in an event.
+     * @param deletionMean the deletion depth of coverage negative shift.
+     * @param duplicationMean the duplication depth of coverage positive shift.
+     * @param meanEventTargetDistance the expectation of the distance between consecutive targets in an event.
+     *
+     * @throws IllegalArgumentException if any of the model parameters has an invalid value.
+     */
+    public CopyNumberTriStateHiddenMarkovModel(final double eventRate,
+                                               final double meanNumberOfTargetsPerEvent,
+                                               final double deletionMean,
+                                               final double duplicationMean,
+                                               final double meanEventTargetDistance) {
+        checkParameterRange(VALID_EVENT_RATE_RANGE, eventRate, "event rate");
+        checkParameterRange(VALID_MEAN_NUMBER_OF_TARGETS_PER_EVENT_RANGE, meanNumberOfTargetsPerEvent,
```

Comment:
checkParameterRange does a bit more than isPositive in terms of composing a detailed error message. I prefer to keep it as it is.